azurerm/resource_arm_azure_firewall_network_rule_collection.go

```diff
@@ -0,0 +1,355 @@
+	client := meta.(*ArmClient).azureFirewallsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	name := d.Get("name").(string)
+	resourceGroup := id.ResourceGroup
+	firewallName := id.Path["azureFirewalls"]
+
+	azureRMLockByName(firewallName, azureFirewallResourceName)
+	defer azureRMUnlockByName(firewallName, azureFirewallResourceName)
+
+	firewall, err := client.Get(ctx, resourceGroup, firewallName)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+	_, exists := findArmAzureFirewallNetworkRuleCollectionByName(&firewall, name)
+	if !exists {
+		return nil
+	}
+	updatedCollections := removeArmAzureFirewallNetworkRuleCollectionByName(&firewall, name)
+	firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections = updatedCollections
+
+	ipConfigs := fixArmAzureFirewallIPConfiguration(&firewall)
+	firewall.AzureFirewallPropertiesFormat.IPConfigurations = &ipConfigs
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, firewallName, firewall)
+	if err != nil {
+		return fmt.Errorf("Error creating/updating Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	err = future.WaitForCompletion(ctx, client.Client)
+	if err != nil {
+		return fmt.Errorf("Error waiting for creation/update of Azure Firewall %q (Resource Group %q): %+v", firewallName, resourceGroup, err)
+	}
+
+	return nil
+}
+
+func expandArmAzureFirewallNetworkRuleCollection(d *schema.ResourceData) network.AzureFirewallNetworkRuleCollection {
+	name := d.Get("name").(string)
+	action := network.AzureFirewallRCActionType(d.Get("action").(string))
+	rules := expandArmAzureFirewallNetworkRules(d)
+	properties := network.AzureFirewallNetworkRuleCollectionPropertiesFormat{
+		Action: &network.AzureFirewallRCAction{
+			Type: action,
+		},
+		Priority: utils.Int32(int32(d.Get("priority").(int))),
+		Rules:    &rules,
+	}
+	col := network.AzureFirewallNetworkRuleCollection{
+		Name: &name,
+		AzureFirewallNetworkRuleCollectionPropertiesFormat: &properties,
+	}
+	return col
+}
+
+func findArmAzureFirewallNetworkRuleCollectionByName(firewall *network.AzureFirewall, name string) (*network.AzureFirewallNetworkRuleCollection, bool) {
+	for _, collection := range *firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections {
+		if collection.Name != nil && *collection.Name == name {
+			return &collection, true
+		}
+	}
+	return nil, false
+}
+
+func removeArmAzureFirewallNetworkRuleCollectionByName(firewall *network.AzureFirewall, name string) *[]network.AzureFirewallNetworkRuleCollection {
+	var index int
+	collections := *firewall.AzureFirewallPropertiesFormat.NetworkRuleCollections
+	for i, collection := range collections {
+		if collection.Name != nil && *collection.Name == name {
+			index = i
```

Comment:
I think it may also be possible to call `delete(collections, collection)` here, since we're breaking anyway?